### ISSUE TITLE
Possible to allow access to component's `this` in chart event handlers (or other places)?

### ISSUE THREAD

**Issue by knod** [whawker/react-jsx-highcharts#188]
#### Bug or Feature Request?
Maybe a feature request. Maybe already exists. Maybe impossible.

#### Description
We can't use arrow functions for event handlers because we need the chart's `this` to access other things. That means that we don't have access to `this.props` in there, etc. Right now, I've found two workarounds.

1. One way:
```js
prepChart = (chart) => {
  this.chart = chart;
  // Couldn't seem to just add it as `this.chart.myProps`
  this.chart.options.myProps = this.props;
};

formatLabel = function () {
  // It's actually more complex stuff in here, but this
  // is just an example of what you have to do to access props
  var prefix = this.chart.options.myProps.prefix,
      val    = this.axis.defaultLabelFormatter.call(this);
  return prefix + val;
};

render () {
  // ...
  return (
    <HighchartsChart callback = { this.prepChart }>
      // ...
      <XAxis labels = {{ formatter: this.formatLabel }}>
  );
}
```

2. Another way:
```js
formatLabel = function (chartComponent) {
  // It's actually more complex stuff in here, but this
  // is just an example of what you have to do to access props
  var prefix = chartComponent.props.prefix,
      val    = this.axis.defaultLabelFormatter.call(this);
  return prefix + val;
};

render () {
  // ...
  var chartComponent = this;
  // ...
  return (
    <XAxis labels = {{ formatter: function () {
      return (chartComponent.formatLabel.bind(this))(chartComponent) }}>
  );
}
```

2b. [ Edit: That last one can also be done like this:
```js
formatLabel = function (chartComponent) {
  // It's actually more complex stuff in here, but this
  // is just an example of what you have to do to access props
  var prefix = chartComponent.props.prefix,
      val    = this.axis.defaultLabelFormatter.call(this);
  return prefix + val;
};

render () {
  // ...
  var chartComponent = this,
      formatLabel    = function () {
        return (chartComponent.formatLabel.bind(this))(chartComponent);
      };
  // ...
  return (
    <XAxis labels = {{ formatter: formatLabel }}>
  );
}
```
] (end edit)

3. Not sure it's possible, but what might be more useful is something like:
```js
formatLabel = function () {
  // It's actually more complex stuff in here, but this
  // is just an example of what you'd do to access props
  var prefix = this.chart.component.props.prefix,
      val    = this.axis.defaultLabelFormatter.call(this);
  return prefix + val;
};

render () {
  return (
    // ...
    <XAxis labels = {{ formatter: this.formatLabel }}>
  );
}
```

#### Versions
* React JSX Highcharts version: ^3.3.1-alpha.1 (why did npm install alpha, btw?)
* Highcharts version: ^6.2.0
* React version: 16.3.2

**Comment by anajavi** [comment]
```this``` could be wrapped in react-jsx-highcharts and passed as parameter to formatter function. But I think it would be much better to fix it on Highcharts side of things.

**Comment by anajavi** [comment]
Something like this could also work (not tested)
```jsx
constructor(props) {
  super(props);
  this.formatLabel = createFormatLabel(this);
}
createFormatLabel = (self) => {
  return function formatLabel() {
    // do some things with self and this
  }
}
render() {
  // ...
  <XAxis labels = {{ formatter: this.formatLabel }}>
}
```

**Comment by knod** [comment]
@anajavi 
> better to fix it on Highcharts side of things.

Welp, that led me to discover that Highcharts _does_ pass the `this` as an argument to the function :P 

I wrote a whole thing! At least that solves the problem, though. I do wish they had that in their docs. That would have saved a lot of time. Maybe it's there somewhere and I missed it. Le sigh.

~Hmm, yeah. It might be better if Highcharts would pass the `this` instead of changing the context. Not sure how long it'd be till they processed a request like that and made the change. That would break a lot of past code and would require a new major version :/~

~Could we somehow use that pattern, or something like it, for what react-jsx-highcharts does? Creating an extra parameter for the formatter function? Use of it might look something like my number 2b (I've numbered them now), but with a bit less work for... well, for me.~

~Though if Highcharts then did change _their_ API to pass an argument, that would break~

> Something like this could also work (not tested)

I'm going to go back and use the Highcharts argument now, but I did like your version in some ways. I work with a lot of people new to coding, though, and they struggle with the idea of a function that returns a function.
